src/PIL/ImageGrab.py

```diff
@@ -135,6 +135,12 @@ def grabclipboard():
     else:
         if shutil.which("wl-paste"):
             args = ["wl-paste"]
+            output = subprocess.check_output(["wl-paste", "-l"]).decode()
+            mime_types = output.splitlines()
+            for image_type in ["image/gif", "image/png"]:
+                if image_type in mime_types:
+                    args.extend(["-t", image_type])
```

Comment:
```suggestion
            clipboard_mimetypes = output.splitlines()
            Image.preinit()
            for mimetype in Image.MIME.values():
                if mimetype in clipboard_mimetypes:
                    args.extend(["-t", mimetype])
```
This would allow you to use more than just GIF and PNG.